conans/model/ref.py

```diff
@@ -287,6 +287,11 @@ def __repr__(self):
     def __str__(self):
         return "%s:%s" % (self.ref, self.id)
 
+    def __lt__(self, other):
+        me = self.ref, self.id, self.revision or ""
+        other = other.ref, other.id, other.revision or ""
+        return me < other
```

Comment:
👍  ...but I wonder why we need an `lt` operator now...

Reply:
This is the first original bug. The ``sorted(prefs)`` that is being done to compute the ``sha`` = package_id crashes, because it is trying to order packages with PREV and other without PREV